src/main/java/com/github/manolo8/darkbot/core/api/GameAPIImpl.java

```diff
@@ -51,6 +53,8 @@
 
     protected long lastFailedLogin;
 
+    protected AbstractDataReader[] dataReaders = new AbstractDataReader[10];
```

Comment:
```suggestion
    protected DataReader[] dataReaders = new DataReader[10];
```
Can't we simply rely in *whatever* data reader this happens to be? that's the point of the interface, not relying on the impl